src/win32/git2.rc

```diff
@@ -29,9 +40,9 @@ BEGIN
     BEGIN
       VALUE "FileDescription",	"libgit2 - the Git linkable library\0"
       VALUE "FileVersion",	LIBGIT2_VERSION "\0"
-      VALUE "InternalName",	LIBGIT2_FILENAME ".dll\0"
+      VALUE "InternalName",	STRINGIZE_(LIBGIT2_FILENAME) ".dll\0"
       VALUE "LegalCopyright",	"Copyright (C) the libgit2 contributors. All rights reserved.\0"
-      VALUE "OriginalFilename",	LIBGIT2_FILENAME ".dll\0"
+      VALUE "OriginalFilename",	STRINGIZE_(LIBGIT2_FILENAME) ".dll\0"
```

Comment:
Why do we need to stringize only _this_ value?  Curious what is compiled in here using gcc.  The easiest way to see this is probably the details tab on the properties window for the resulting DLL (in Windows Explorer).  I'd love to see what it shows.

Reply:
The reason why we need to stringize it at all is because `windres.exe` doesn't understand the concatenation syntax if we don't. I believe this is because this value is defined by CMake  and apparently doesn't contain quotes when expanded.

It also seems to be the only value used here that is only defined by CMake and not defined anywhere in the code (except at the top of this same file), that's why it's the only one to stringize.

If I change the line `#ifndef LIBGIT2_FILENAME` to `#if !defined(LIBGIT2_FILENAME) && defined(GCC_WINDRES)`, then the macro is set by code, it has quotes and `windres.exe` is happy. I could undef the value for windres at the very top, but I wanted to preserve the custom library name.

Here's the Details tab of the compiled binary (this is v1.0.1 but .rc file is the same):

![image](https://user-images.githubusercontent.com/56644701/84802153-7a813c80-b000-11ea-89ba-4d9a13a1d0b2.png)

Reply:
Got it - thanks for walking me through this - another read through of everything and it all made sense to me.  😁